generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/fieldselectionstrategy/FieldAppearanceAnalyser.java

```diff
@@ -0,0 +1,34 @@
+package com.scottlogic.deg.generator.walker.reductive.fieldselectionstrategy;
+
+import com.scottlogic.deg.common.profile.Field;
+import com.scottlogic.deg.common.profile.constraints.atomic.AtomicConstraint;
+import com.scottlogic.deg.generator.decisiontree.ConstraintNode;
+import com.scottlogic.deg.generator.decisiontree.visualisation.BaseVisitor;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class FieldAppearanceAnalyser extends BaseVisitor {
+
+    Map<Field, Integer> fieldAppearances = new HashMap<>();
+
+    @Override
+    public ConstraintNode visit(ConstraintNode constraintNode){
+        constraintNode.getAtomicConstraints().stream()
+            .map(AtomicConstraint::getField)
+            .distinct()
+            .forEach(this::countField);
+        return constraintNode;
+    }
+
+    private void countField(Field field) {
+        if (fieldAppearances.containsKey(field)){
```

Comment:
fieldAppearances.compute(field, (k, v) -> v == null ? 1 : v++);

Reply:
@r-stuart turns out its
fieldAppearances.compute(field, (k, v) -> v == null ? 1 : v+1);
not 
fieldAppearances.compute(field, (k, v) -> v == null ? 1 : v++);

Reply:
`++v` would also work, but I wouldn't endorse it :stuck_out_tongue: